sdk/src/asset_handlers/bmff_io.rs

```diff
@@ -1257,4 +1386,27 @@ pub mod tests {
         }
         assert!(success)
     }
+
+    #[test]
+    fn test_remove_c2pa() {
+        let source = fixture_path("video1.mp4");
+
+        let mut success = false;
+        if let Ok(temp_dir) = tempdir() {
+            let output = temp_dir_path(&temp_dir, "mp4_test.mp4");
+
+            if let Ok(_size) = std::fs::copy(&source, &output) {
+                let bmff = BmffIO::new("mp4");
+
+                if let Ok(()) = bmff.remove_cai_store(&output) {
+                    match bmff.read_cai_store(&output) {
+                        Ok(_) => success = false,
```

Comment:
Assigning to `success` and matching makes it a bit hard to follow the overall control flow.

Perhaps you could use `if let ... else` and have more specific assertions at the point where each failure case is detected?